### PR TITLE
simplifying logic since CHUNK2 is multiple of RATE now

### PR DESCRIPTION
**Pull request by roman3017**

### CODE REVIEW

File: bin/doppler-gestures.py [hunk 1]
```diff
@@ -43,29 +43,24 @@ def tonePlayer(freq, sync):
...
 
     p = pyaudio.PyAudio()
-
-    A = float(2**15 - 1)
-    CHUNK2 = ((CHUNK+RATE-1)/RATE)*RATE
-
     stream = p.open(format=pyaudio.paInt16,
                     channels=2,
                     rate=RATE,
-                    frames_per_buffer=CHUNK2,
```
Comment (cwoodall) [inline bin/doppler-gestures.py:53]: Good change here.

File: bin/doppler-gestures.py [hunk 2]
```diff
@@ -16,7 +16,7 @@
 from scipy import signal
 
 TONE = 20000
-WINDOW = 500
+WINDOW = 1000
```
Comment (cwoodall) [inline bin/doppler-gestures.py:19]: Why did this change?

File: bin/doppler-gestures.py [hunk 3]
```diff
@@ -43,29 +43,24 @@ def tonePlayer(freq, sync):
...
+                    frames_per_buffer=RATE,
                     output=True,
                     input=False)
 
     stream.start_stream()
     sync.set()
-    s = 0
-    j = 0
+
+    A = float(2**15 - 1)
```
Comment (cwoodall) [inline bin/doppler-gestures.py:56]: This should be defined as a constant somewhere. I want to actually fix up this whole setup and maybe even make some generic wrapper for pyaudio for doing threaded "realish time" audio processing

File: pydoppler/ambiguity.py [hunk 4]
```diff
@@ -32,8 +32,8 @@ def ambiguity(code, nfreq=1):
     """
     inlen = len(code)
     outlen = 2*inlen - 1
-    if nfreq < inlen:
```
Comment (cwoodall) [inline pydoppler/ambiguity.py:35]: What does the ambiguity code do? I am a little confused at the moment.

**Comment by cwoodall** [comment]
I am merging it, but I am curious about the ambiguity code.

**Comment by roman3017** [comment]
there are many resources on the web on ambiguity function like:

https://en.wikipedia.org/wiki/Ambiguity_function
http://www.mathworks.com/help/phased/examples/waveform-analysis-using-the-ambiguity-function.html

it is 2d array, which does autocorrelation of each row and each row is mixed with different frequency. so rows should indicate distance from object and columns speed. it is being used in radars and sonars

File: bin/doppler-gestures.py (hunk 2 above)
Comment (roman3017) [inline bin/doppler-gestures.py:19]: this particular change was not necessary, i just ended up using 1000 each time for the plotter view and 512 for ambiguity view, which works with chunk 256/512

**Comment by roman3017** [comment]
one can see the signal time delay from the autocorrelation and hence can infer a distance. on the other hand a frequency shift indicates a doppler effect and one can infer a speed.

**Comment by roman3017** [comment]
but i am not sure if it is applicable to our signal, it is usually used with a specific signal patterns and bursts. but i thought it would be interesting to play with it.

**Comment by cwoodall** [comment]
I would like to change the signal pattern to bursts/chirps at some point. CWFM radar or pulse doppler. my main concern is whether or not one can do this with their laptop easily/generalizable. The microphones can be in different places and the math changes depending on this, but I think you could have a calibration phase where you stand still then min max it. Also if you use different frequencies/burst times for R and L you can probably do some pretty interesting stuff with getting more 2d information. Since those speakers tend to be on different sides of the laptop.

**Pull request merged**